### PR TITLE
Apply retry patch

### PR DESCRIPTION
**Pull request by cloutierMat**
# Motivation

When running aws terraform locally against Localstack, a lot of the resources end up having a delay before being checked which artificially extends `apply` and `destroy` of resources. As far as I can tell all resources are calling the `StateChangeConf.WaitForStateContext` from the terraform plugin sdk helpers.

# Changes

Created a script to apply patches to reset all delay and polling interval to defaults. The default being an exponential backoff retry. It will double wait time until a max of 10 seconds wait time.

```
0 ms
100 ms
200 ms
400 ms
...
10 s
10 s
```

### CODE REVIEW

File: create_all_tags.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import os
+from git import Repo
+
+repo = Repo(".")
+
+
+KNOWN_LOCATIONS_OF_STATE_GO = ("helper/resource/state.go", "helper/retry/state.go")
```
Comment (gregfurman) [inline create_all_tags.py:7]: Nice that you accounted for that `resource/helper` as well!

File: create_all_tags.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        )
+        if user_input != "":
+            repo.git.checkout(f"tags/{tag} {file_path}".split())
+            continue
+
+        repo.git.add(file_path)
+        repo.git.commit(f"-m 'creating tag for version {tag}'")
+        repo.git.tag(f"-d {tag}".split())
+        repo.git.tag(tag)
+        repo.git.push(f"origin tag {tag}".split())
```
Comment (gregfurman) [inline create_all_tags.py]: Wondering if `git push origin --tags` after the for-loop would work better. That way, each patch isn't delayed by pushing to remote.

File: create_all_tags.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+from git import Repo
+
+repo = Repo(".")
+
+
+KNOWN_LOCATIONS_OF_STATE_GO = ("helper/resource/state.go", "helper/retry/state.go")
+KNOWN_FUNC_DECLARATION = (
+    "func (conf *StateChangeConf) WaitForStateContext(ctx context.Context) (interface{}, error) {\n",
+)
+PATCH = [
```
Comment (gregfurman) [inline create_all_tags.py:11]: This is absolutely brilliant hahaha. Genuinely love this for a hackathon 🤣

**Review by gregfurman — APPROVED** [review]
LGTM -- 1 small comment on the tag pushing but otherwise think this is great lol

File: create_all_tags.py (hunk 2 above)
Comment (cloutierMat) [reply create_all_tags.py]: Good idea, I have a couple more enhancement to the way I am using gitpython to push as well. I will add that to the list of improvements

**Pull request merged**